src/compiler/parser.ts

```diff
@@ -8053,6 +8053,22 @@ namespace ts {
                     return token() === SyntaxKind.OpenBraceToken ? parseJSDocTypeExpression() : undefined;
                 }
 
+                function parseBracketNameInTemplateTag(): { name: Identifier, defaultType?: TypeNode } {
+                    const isBracketed = parseOptionalJsdoc(SyntaxKind.OpenBracketToken);
+                    if (isBracketed) {
+                        skipWhitespace();
+                    }
+                    const name = parseJSDocIdentifierName(Diagnostics.Unexpected_token_A_type_parameter_name_was_expected_without_curly_braces);
+                    let defaultType: TypeNode | undefined;
+                    if (isBracketed) {
+                        skipWhitespace();
+                        parseExpected(SyntaxKind.EqualsToken);

```

Comment:
Silly question: what would just `[T]` mean?

Probably nothing, but I was just thinking about the way the parser allows almost any combination of characters for existing JSDoc syntax.

Reply:
`[T]` is a syntax error, and just means `T`. For the bracketed syntax to work for a `@template` tag you *must* specify `=type`